Revise s_client and s_server verbiage re secure renegotiation. (#16937)
openssl/openssl: CHANGES.md

### Changes between 3.0 and 3.1 [xx XXX xxxx]

 * s_client and s_server apps now explicitly say when the TLS version

>    obviates a need for renegotiation. This avoids confusion between

Reviewer: ```suggestion
   does not include the renegotiation mechanism. This avoids confusion between
```

Author: done

   between that scenario versus when the TLS version includes secure
   renegotiation but the peer lacks support for it.

> 

Reviewer 2: Please add your name here similarly to other entries.

Author: Done.
